src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/AbstractReferenceHandler.cs

```diff
@@ -68,32 +71,73 @@ internal async Task<List<ProjectSystemReferenceInfo>> GetReferencesAsync(Configu
 
         private static async Task<bool> GetAttributeTreatAsUsedAsync(IProjectProperties metadata)
         {
+            var propertyNames = await metadata.GetPropertyNamesAsync();
             string? value = await metadata.GetEvaluatedPropertyValueAsync(ProjectReference.TreatAsUsedProperty);
 
             return value != null && PropertySerializer.SimpleTypes.ToValue<bool>(value);
         }
 
-        internal async Task<bool> UpdateReferenceAsync(ConfiguredProject selectedConfiguredProject, ProjectSystemReferenceUpdate referenceUpdate, CancellationToken cancellationToken)
+        private async Task<IProjectItem> GetProjectItems(ConfiguredProject selectedConfiguredProject,
```

Comment:
A few issues with this method signature:

1. It can return `null` so should reflect that in the return type
2. It is async so should have an `Async` suffix
3. It returns a single item. The name suggests it returns multiple.

Instead:

```suggestion
        private async Task<IProjectItem?> GetProjectItemAsync(ConfiguredProject selectedConfiguredProject,
```